File: 94.cpp
/*
二叉树的中序遍历
给定一个二叉树的根节点 root ，返回 它的 中序 遍历 。
*/
using namespace std;
#include<vector>
#include<iostream>
#include<algorithm>
#include<numeric>
#include<unordered_map>
#include<math.h>
#include<queue>
#include<stack>

struct TreeNode {
    int val;
    TreeNode *left;
    TreeNode *right;
    TreeNode() : val(0), left(nullptr), right(nullptr) {}
    TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
    TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
};

class Solution {
public:
    vector<int> inorderTraversal(TreeNode* root) {
        vector<int> out;
    }
};

int main()
{
    TreeNode *root = new TreeNode(1);
    root->right = new TreeNode(2);
    root->right->left = new TreeNode(3);

	Solution sol;
	vector<int> out = sol.inorderTraversal(root);
    for(auto num : out) {
        cout << num << endl;
    }
	cout << endl;
	system("pause");
}

File: leetcode_py/94.py
from typing import List, Optional
import sys

class TreeNode:
    def __init__(self, val=0, left=None, right=None):
        self.val = val
        self.left = left
        self.right = right

class Solution:
    def inorderTraversal(self, root: Optional[TreeNode]) -> List[int]:
        if root == None:
            return []
        out = []
        out.extend(self.inorderTraversal(root.left))
        out.append(root.val)
        out.extend(self.inorderTraversal(root.right))
        return out

if __name__ == '__main__':
    root = TreeNode(1)
    root.right = TreeNode(2)
    root.right.left = TreeNode(3)
    sol = Solution()
    out = sol.inorderTraversal(root)
    print(out)